File: 50-web-development/20-javascript-typescript/00-example/src/main/typeScript/classes/Person.ts
export class Person {

    public constructor(private firstName: string, private lastName?: string, private hobby = 'cycling') {

    }

    private getFullName(): string {
        return this.firstName + ' ' + this.lastName;
    }

    public printName(): void {
        console.log(this.getFullName());
    }
}

let bob: Person = new Person('bob', 'singing');
let billie: Person = new Person('billie', 'eilish');

File: 10-programming-fundamentals-java/20-oop/10-objects-and-classes/21-classes/src/main/java/basic/codelab03/Person.java
package basic.codelab03;

public class Person {
    private String firstName;
    private String lastName;
    private int age;

    public Person(int age) {
        this.age = age;
    }

    public Person(String firstname, String lastname, int age) {
        this.firstName = firstname;
        this.lastName = lastname;
        this.age = age;
    }

    public int calculateAllowance() {
        if (0 < age && age <= 12) {
            return 100;
        }
        if (age < 18) {
            return (100 + (age - 12) * 5);
        }
        return 0;
    }
}
